walg_archive.c

```diff
@@ -101,7 +124,19 @@ walg_archive_configured(void)
 static bool 
 walg_archive_file(const char *file, const char *path) 
 {	
-	if (send(fd, file, strlen(file)*sizeof(char), 0 ) == -1)
+	char message_type[] = "F";
+	uint8 message_len = sizeof(file) + 1;
+
+	char *p = palloc(sizeof(char)*message_len);
+	p[0] = message_type[0];
+	p[1] = message_len;
+	
+	for (int i = 2; i < message_len; i++) 
+	{
+		p[i] = file[i-2];
+	}
+
+	if (send(fd, p, sizeof(p), 0 ) == -1)
```

Comment:
`sizeof(p)` would return the size of the pointer. You need to provide the actual message length here.